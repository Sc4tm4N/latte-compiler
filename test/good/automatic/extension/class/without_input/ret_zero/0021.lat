
class node {
  int node;

  int node() {
     return node;
  }
}

class node2 extends node {
  int node2;

  int node() {
    return node * node2;
  }
}

class node3 extends node2 {
  int node3;

  int node() {
    return node * node2 * node3;
  }
}

int main() {
  node3 node = new node3;
  node.node3 = 33;
  node.node2 = 22;
  node.node = 11;

  printInt(node.node3);
  printInt(((node2) node).node2);
  printInt(((node)((node2) node)).node);

  printInt(node.node());
  printInt(((node2) node).node());
  printInt(((node)((node2) node)).node());

  printInt(node.node());
  printInt(((node2) node).node());
  printInt(((node)((node2) node)).node());

  return 0;
}
