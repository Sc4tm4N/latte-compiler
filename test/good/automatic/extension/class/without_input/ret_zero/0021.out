33
22
11
7986
7986
7986
7986
7986
7986
